packages/Webkul/Shop/src/Http/Controllers/API/CartController.php

```diff
@@ -229,15 +229,21 @@ public function destroyCoupon(): JsonResource
      */
     public function crossSellProducts()
     {
-        $productIds = Cart::getCart()->items->pluck('product_id')->toArray();
+        try {
```

Comment:
Instead of wrapping this in try/catch statement, we can check for the cart if it's empty or not. If empty then send a blank array response.

Reply:
Yes, indeed. It was my first solution for early return, but when I scan the other method in this file, most of the method is used to handle exceptions with try catch, and I just want to make it uniform.
```
public function crossSellProducts()
    {
        $cart = Cart::getCart();

        if (! $cart) {
            return new JsonResource([
                'data' => []
            ]);
        }

        $productIds = $cart->items->pluck('product_id')->toArray();

        $products = $this->productRepository
            ->select('products.*', 'product_cross_sells.child_id')
            ->join('product_cross_sells', 'products.id', '=', 'product_cross_sells.child_id')
            ->whereIn('product_cross_sells.parent_id', $productIds)
            ->groupBy('product_cross_sells.child_id')
            ->paginate();

        return ProductResource::collection($products);
    }
```

Reply:
Yeah, because other methods are not GET requests and we are handing the error response on the front as well for NOn-GET requests.

Reply:
Any update on this ?

Reply:
Any update on this?

Reply:
@phyothiha Please check the comments mentioned above.